kafka/src/main/java/io/micronaut/configuration/kafka/annotation/ErrorStrategy.java

```diff
@@ -61,13 +61,27 @@
     String retryDelay() default DEFAULT_DELAY_IN_SECONDS + "s";
 
     /**
-     * The retry count used with RETRY_ON_ERROR and RETRY_EXPONENTIALLY_ON_ERROR
+     * The fixed retry count used with RETRY_ON_ERROR and RETRY_EXPONENTIALLY_ON_ERROR
      * {@link io.micronaut.configuration.kafka.annotation.ErrorStrategyValue}.
      *
+     * <p>{@code retryCount} will be overridden by {@code retryCountValue} if they are both set.
+     *
      * @return the retry count of how many attempts should be made
+     * @see ErrorStrategy#retryCountValue()
      */
     int retryCount() default DEFAULT_RETRY_COUNT;
 
+    /**
+     * The dynamic retry count used with RETRY_ON_ERROR and RETRY_EXPONENTIALLY_ON_ERROR
+     * {@link io.micronaut.configuration.kafka.annotation.ErrorStrategyValue}.
+     *
+     * <p>{@code retryCountValue} takes precedence over {@code retryCount} if they are both set.
+     *
+     * @return the retry count of how many attempts should be made
+     * @see ErrorStrategy#retryCount()
+     */
+    String retryCountValue() default "";
```

Comment:
This needs to be an `@AliasFor`

Reply:
Good catch! I annotated those two as an alias for each other, and now I don't need to manually get one if the other one is absent.

The only catch is that Micronaut seems to favor `int` over `String` when they are both set, so I had to update the documentation to make clear that `retryCount` will take preference over `retryCountValue`.